### PR TITLE
add github action to update submodules

### PR DESCRIPTION
**Pull request by habibayassin**
Signed-off-by: habibayassin <habibayassin@aucegypt.edu>

### CODE REVIEW

**Comment by habibayassin** [comment]
merged the changes on my fork's master and triggered the workflow and it created a draft PR: https://github.com/habibayassin/OpenROAD-flow-scripts/pull/14

**Comment by maliberty** [comment]
Yosys releases once a month with a new tag.  I think it would be better to use the latest tag rather than master.

**Comment by habibayassin** [comment]
> Yosys releases once a month with a new tag. I think it would be better to use the latest tag rather than master.

@maliberty when the fork of Yosys was updated by pulling upstream changes in master the tag was not affected so I don't think we will be able to pull the latest tag

**Comment by maliberty** [comment]
Use git pull --tags to update the tags as well

**Comment by habibayassin** [comment]
@vvbandeira @maliberty I fetched the latest tags in the forked yosys and the GHA is using the latest which is 0.23 not 0.22

**Comment by habibayassin** [comment]
commits need to be squashed before merge

**Comment by maliberty** [comment]
lgtm but I'll leave it to @vvbandeira to approve

File: .github/workflows/github-actions-cron-update-OR.yml
Comment (vvbandeira) [inline .github/workflows/github-actions-cron-update-OR.yml]: Where did the `41898282` came from?

File: .github/workflows/github-actions-cron-update-OR.yml
Comment (vvbandeira) [inline .github/workflows/github-actions-cron-update-OR.yml:24]: Are we sure on the order here? Should we do a `fetch` before setting the `has_update` value?

File: .github/workflows/github-actions-cron-update-yosys.yml
Comment (habibayassin) [inline .github/workflows/github-actions-cron-update-yosys.yml]: @vvbandeira do you think I should check the difference between the head and the latest tag not the master? or it wouldn't matter?

Comment (vvbandeira) [reply .github/workflows/github-actions-cron-update-yosys.yml]: since we are going to update to the tag (going from Matt's comment earlier) it makes sense

**Review by vvbandeira — APPROVED** [review]

**Pull request merged**
